### DISCUSSION TITLE
Help with a Rule

### DISCUSSION THREAD

**Discussion by dali-dbr** [arendst/Tasmota#20808] [Q&A]
Hi.

Can anyone help me create a rule where the sensor changes to immediately send an MQTT payload?

![image](https://github.com/arendst/Tasmota/assets/6317229/b61e8d84-cb1e-4668-a70c-c0c2f4e7e07b)

For Example:
**WHEN** LD2410 Moving Distance changes **DO** instantly publish MQTT update

Thank you very much

**Comment by barbudor** [comment]
This one is a little more than what you are asking for but probably best to use a threashold
https://tasmota.github.io/docs/Rules/#transmit-sensor-value-only-when-a-delta-is-reached

**Reply by sfromis** [reply]
Note that to reference fields from the JSON with 3 values in the same array, the trigger reference would be `LD2410#Distance[1]` for the first value.

**Reply by dali-dbr** [reply]
Thank you.

This is exactly what I was testing just now.

```
Rule1
ON LD2410#Distance[1]>%var1% DO Backlog var1 %value%;  publish tele/199-mmWave/SENSOR {"Time":"%TIMESTAMP%","LD2410":{"Distance":[%value%,0.0,0.0],"Energy":[0,0]}}; var2 %value%; add1 0.5; sub2 0.5 ENDON
ON LD2410#Distance[1]<%var2% DO Backlog var2 %value%;  publish tele/199-mmWave/SENSOR {"Time":"%TIMESTAMP%","LD2410":{"Distance":[%value%,0.0,0.0],"Energy":[0,0]}}; var1 %value%; add1 0.5; sub2 0.5 ENDON
```
This rule seems to work well. As it is now it works only for the one value (currently [1] - LD2410 moving Distance), I can change the rule to other two values if needed, and no need to monitor multiple values.

Thank you

Br,
Dali

**Reply by sfromis** [reply]
The way you are doing it there, mimicking standard payload format, the subscriber will get pretty broken values with both energy and the 2nd and 3rd distance value. If you instead execute a `Status 10` command, Tasmota will publish the full sensor payload.

**Comment by dali-dbr** [comment]
> If you instead execute a `Status 10` command, Tasmota will publish the full sensor payload.

I'm not sure I follow you here. Please keep in mind I did some basic Tasmota things, but nothing as advanced as writing rules. 

typing `status 10` in Tasmota console will output this:
`23:54:57.235 MQT: stat/199-mmWave/STATUS10 = {"StatusSNS":{"Time":"2024-02-25T23:54:57","LD2410":{"Distance":[0.0,0.0,0.0],"Energy":[0,0]}}}`

but will not publish MQTT


I added %topic% variable to the rule so I can copy the rule to other devices.
```
Rule1
ON LD2410#Distance[1]>%var1% DO Backlog var1 %value%;  publish tele/%topic%/SENSOR {"Time":"%TIMESTAMP%","LD2410":{"Distance":[%value%,0.0,0.0],"Energy":[0,0]}}; var2 %value%; add1 0.5; sub2 0.5 ENDON
ON LD2410#Distance[1]<%var2% DO Backlog var2 %value%;  publish tele/%topic%/SENSOR {"Time":"%TIMESTAMP%","LD2410":{"Distance":[%value%,0.0,0.0],"Energy":[0,0]}}; var1 %value%; add1 0.5; sub2 0.5 ENDON
```

**Reply by sfromis** [reply]
That `Status 10` command **is** publishing the result to MQTT, that's indicated by the `MQT:` prefix. And you can see the topic it is publishing to after that. Not the same topic as for telemetry period, but that makes sense and that is not what it is.

**Comment by dali-dbr** [comment]
> That `Status 10` command **is** publishing the result to MQTT, that's indicated by the `MQT:` prefix. And you can see the topic it is publishing to after that. Not the same topic as for telemetry period, but that makes sense and that is not what it is.

Thank you for your reply.

When I type status 10 in in Tasmota console I get this in output in the Tasmota console.
`08:12:40.026 MQT: stat/199-mmWave/STATUS10 = {"StatusSNS":{"Time":"2024-02-26T08:12:40","LD2410":{"Distance":[0.0,0.0,0.0],"Energy":[0,0]}}}`

No MQTT is sent (I'm using MQTT Explorer to monitor). I'm not sure how you are trying to explain the actual use of Status 10. Can you give me an example?

Thank you

**Reply by sfromis** [reply]
Again, it **is** publishing via MQTT, you just have to subscribe to the topic it is using, in this case at least `stat/199-mmWave/STATUS10`
